vital/tests/test_local_cartesian.cxx

```diff
@@ -0,0 +1,129 @@
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * \file
+ * \brief core geo_point class tests
+ */
+
+#include <test_eigen.h>
+#include <test_gtest.h>
+
+#include <vital/types/geodesy.h>
+#include <vital/types/local_cartesian.h>
+
+#include <vital/plugin_loader/plugin_manager.h>
+
+static auto constexpr wgs = kwiver::vital::SRID::lat_lon_WGS84;
+static auto constexpr crs_utm_18n = kwiver::vital::SRID::UTM_WGS84_north + 18;
+
+static auto const origA = kwiver::vital::geo_point(kwiver::vital::vector_3d( -73.759291, 42.849631, 0 ), wgs);
+static auto const origB = kwiver::vital::geo_point(kwiver::vital::vector_3d( 601375.01, 4744863.31, 0 ), crs_utm_18n);
+static auto const offset1 = kwiver::vital::vector_3d{ 25, 55, 0 };
+static auto const offset2 = kwiver::vital::vector_3d{ 250, 5500, 50 };
+static auto const geo1 = kwiver::vital::geo_point(kwiver::vital::vector_3d( -73.75898515, 42.85012609, 0 ), wgs);
+static auto const geo2 = kwiver::vital::geo_point(kwiver::vital::vector_3d( -73.75623008, 42.89913984, 52.381 ), wgs);
+
+// ----------------------------------------------------------------------------
+int
+main(int argc, char* argv[])
+{
+  ::testing::InitGoogleTest( &argc, argv );
+  return RUN_ALL_TESTS();
+}
+
+// ----------------------------------------------------------------------------
+TEST(geo_point, constructor)
+{
+  kwiver::vital::local_cartesian lc1(origA);
+  EXPECT_EQ(lc1.get_origin().location(), origA.location());
+  EXPECT_EQ(lc1.get_orientation(), 0);
+
+  kwiver::vital::local_cartesian lc2(origA,33);
+  EXPECT_EQ(lc2.get_origin().location(), origA.location());
+  EXPECT_EQ(lc2.get_orientation(), 33);
+}
+
+// ----------------------------------------------------------------------------
+TEST(geo_point, api)
+{
+  kwiver::vital::local_cartesian lc1(origA);
+  // Test values of the point as originally constructed
+  EXPECT_EQ(lc1.get_origin().location(), origA.location());
+  EXPECT_EQ(lc1.get_orientation(), 0);
+
+  // Modify the location and test the new values
+  lc1.set_origin(geo1, 33);
+  EXPECT_EQ(lc1.get_origin().location(), geo1.location());
+  EXPECT_EQ(lc1.get_orientation(), 33);
+
+  // Modify the location again and test the new values
+  lc1.set_origin(geo2, 22 );
+  EXPECT_EQ(lc1.get_origin().location(), geo2.location());
+  EXPECT_EQ(lc1.get_orientation(), 22);
+}
+
+// ----------------------------------------------------------------------------
+TEST(geo_point, conversion)
+{
+  kwiver::vital::plugin_manager::instance().load_all_plugins();
+
+  kwiver::vital::geo_point geo_out;
+  kwiver::vital::vector_3d cart_out;
+  kwiver::vital::local_cartesian lc_lla(origA);
+
+  // Get the geopoint from an offset
+  lc_lla.convert_from_cartesian(offset1, geo_out);
+  auto const epsilon_geo_out_to_geo1 = (geo_out.location() - geo1.location()).norm();
```

Comment:
consider `lpNorm<Infinity>()` instead of `norm()`.  That gives you the maximum absolution value of the elements.  The L2 distance doesn't make as much sense here.  Also, you really just want to make sure the error is low in all elements, so checking the max is probably the right thing to do.

Reply:
here are the docs:
https://eigen.tuxfamily.org/dox/group__TutorialReductionsVisitorsBroadcasting.html

I would not use `norm()` here because the units on altitude are very different from those on lat and long

Reply:
I just switched the test to test each component individually, seems more flexible and straight forward